components/clp-py-utils/clp_py_utils/start-spider-worker.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+
+"""Script to start multiple Spider workers."""
+
+import argparse
+import logging
+import os
+import pathlib
+import signal
+import socket
+import subprocess
+import sys
+import threading
+
+# Setup logging
+# Create logger
+logger = logging.getLogger("start-spider-worker")
+logger.setLevel(logging.INFO)
+# Setup console logging
+logging_console_handler = logging.StreamHandler()
+logging_formatter = logging.Formatter("%(asctime)s [%(levelname)s] %(message)s")
+logging_console_handler.setFormatter(logging_formatter)
+logger.addHandler(logging_console_handler)
+
+SPIDER_WORKER_TERM_TIMEOUT_SECONDS = 5
+
+stop_event = threading.Event()
+
+
+def handle_sigterm(signum, frame) -> None:
+    """
+    Signal handler for SIGTERM.
+    Stops all running workers gracefully.
+
+    :param signum: The signal number.
+    :param frame: The current stack frame.
+    """
+    logger.info("Received SIGTERM, stopping workers...")
+    stop_event.set()
+
+
+def parse_args() -> argparse.Namespace:
+    """
+    Parses command line arguments.
+    :return: The parsed arguments.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--num-workers", type=int, default=1, help="Number of concurrent Spider workers."
+    )
+    parser.add_argument("--storage-url", type=str, required=True, help="Spider storage URL.")
+    parser.add_argument("--host", type=str, required=False, default=None, help="The worker host.")
```

Comment:
hmm... when i posted https://github.com/y-scope/clp/pull/1606#discussion_r2590752564 , i was just curious why we defined a --host arg parser but didn't specify a value. if we always call start-spider-worker.py with --host, i think we can mark this arg as require=True? Then we don't have to handle the default=None as well